debian/rules

```diff
@@ -2,10 +2,19 @@
 
 BUILDDIR=build_dir
 
-# Ignore false warning with GCC < 11.3: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=99578
-# Reference OpenRCT2 bug report: https://github.com/OpenRCT2/OpenRCT2/issues/16691
-export DEB_CFLAGS_MAINT_APPEND = -Wno-stringop-overflow
-export DEB_CXXFLAGS_MAINT_APPEND = -Wno-stringop-overflow
+# Ignore false warnings with GCC:
+#
+# stringop-overflow false positive in GCC < 11.3
+# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=99578
+#
+# maybe-uninitialized false positive in GCC < 12.3 (note 12.3 is currently unreleased)
+# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105937
+#
+# Reference OpenRCT2 bug reports: 
+# https://github.com/OpenRCT2/OpenRCT2/issues/16691 (-Wno-stringop-overflow)
+# https://github.com/OpenRCT2/OpenRCT2/issues/17371 (-Wno-maybe-uninitialized -Wno-error=restrict -Wno-error=null-dereference)
+export DEB_CFLAGS_MAINT_APPEND   = -Wno-stringop-overflow -Wno-maybe-uninitialized -Wno-error=restrict -Wno-error=null-dereference
+export DEB_CXXFLAGS_MAINT_APPEND = -Wno-stringop-overflow -Wno-maybe-uninitialized -Wno-error=restrict -Wno-error=null-dereference
```

Comment:
I tested it locally (basically: `sudo apt-get install build-essential fakeroot devscripts` then in project root: `debuild -b -uc -us`) and indeed CXX flags are used and I couldn't trigger the restrict/null-dereference on a ubuntu 22.04 system.

```suggestion
export DEB_CXXFLAGS_MAINT_APPEND = -Wno-stringop-overflow -Wno-maybe-uninitialized
```

Reply:
I get this on my Ubuntu 22.10 system (GCC `12.2.0-3ubuntu1`):
```
In file included from /usr/include/c++/12/ios:43,
                 from /usr/include/c++/12/istream:38,
                 from /usr/include/c++/12/fstream:38,
                 from /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:11:
In member function ‘std::basic_streambuf<char, std::char_traits<char> >::egptr() const’,
    inlined from ‘std::basic_streambuf<char, std::char_traits<char> >::sbumpc()’ at /usr/include/c++/12/streambuf:324:49,
    inlined from ‘std::istreambuf_iterator<char, std::char_traits<char> >::operator++()’ at /usr/include/c++/12/bits/streambuf_iterator.h:172:17,
    inlined from ‘std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct<std::istreambuf_iterator<char, std::char_traits<char> > >(std::istreambuf_iterator<char, std::char_traits<char> >, std::istreambuf_iterator<char, std::char_traits<char> >, std::input_iterator_tag)void’ at /usr/include/c++/12/bits/basic_string.tcc:178:6,
    inlined from ‘std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string<std::istreambuf_iterator<char, std::char_traits<char> >, void>(std::istreambuf_iterator<char, std::char_traits<char> >, std::istreambuf_iterator<char, std::char_traits<char> >, std::allocator<char> const&)’ at /usr/include/c++/12/bits/basic_string.h:756:16,
    inlined from ‘ChangelogWindow::GetChangelogText[abi:cxx11]()’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:74:98,
    inlined from ‘ChangelogWindow::ReadChangelogFile()’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:259:46,
    inlined from ‘ChangelogWindow::SetPersonality(int)’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:97:39,
    inlined from ‘WindowChangelogOpen(int) [clone .part.0]’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:317:34:
/usr/include/c++/12/streambuf:493:30: error: potential null pointer dereference [-Werror=null-dereference]
  493 |       egptr() const { return _M_in_end; }
      |                              ^~~~~~~~~
In member function ‘std::basic_streambuf<char, std::char_traits<char> >::gptr() const’,
    inlined from ‘std::basic_streambuf<char, std::char_traits<char> >::sbumpc()’ at /usr/include/c++/12/streambuf:324:33,
    inlined from ‘std::istreambuf_iterator<char, std::char_traits<char> >::operator++()’ at /usr/include/c++/12/bits/streambuf_iterator.h:172:17,
    inlined from ‘std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct<std::istreambuf_iterator<char, std::char_traits<char> > >(std::istreambuf_iterator<char, std::char_traits<char> >, std::istreambuf_iterator<char, std::char_traits<char> >, std::input_iterator_tag)void’ at /usr/include/c++/12/bits/basic_string.tcc:178:6,
    inlined from ‘std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string<std::istreambuf_iterator<char, std::char_traits<char> >, void>(std::istreambuf_iterator<char, std::char_traits<char> >, std::istreambuf_iterator<char, std::char_traits<char> >, std::allocator<char> const&)’ at /usr/include/c++/12/bits/basic_string.h:756:16,
    inlined from ‘ChangelogWindow::GetChangelogText[abi:cxx11]()’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:74:98,
    inlined from ‘ChangelogWindow::ReadChangelogFile()’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:259:46,
    inlined from ‘ChangelogWindow::SetPersonality(int)’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:97:39,
    inlined from ‘WindowChangelogOpen(int) [clone .part.0]’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:317:34:
/usr/include/c++/12/streambuf:490:30: error: potential null pointer dereference [-Werror=null-dereference]
  490 |       gptr()  const { return _M_in_cur;  }
      |                              ^~~~~~~~~
In member function ‘std::basic_streambuf<char, std::char_traits<char> >::gptr() const’,
    inlined from ‘std::basic_streambuf<char, std::char_traits<char> >::sbumpc()’ at /usr/include/c++/12/streambuf:324:33,
    inlined from ‘std::istreambuf_iterator<char, std::char_traits<char> >::operator++()’ at /usr/include/c++/12/bits/streambuf_iterator.h:172:17,
    inlined from ‘std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct<std::istreambuf_iterator<char, std::char_traits<char> > >(std::istreambuf_iterator<char, std::char_traits<char> >, std::istreambuf_iterator<char, std::char_traits<char> >, std::input_iterator_tag)void’ at /usr/include/c++/12/bits/basic_string.tcc:205:6,
    inlined from ‘std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string<std::istreambuf_iterator<char, std::char_traits<char> >, void>(std::istreambuf_iterator<char, std::char_traits<char> >, std::istreambuf_iterator<char, std::char_traits<char> >, std::allocator<char> const&)’ at /usr/include/c++/12/bits/basic_string.h:756:16,
    inlined from ‘ChangelogWindow::GetChangelogText[abi:cxx11]()’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:74:98,
    inlined from ‘ChangelogWindow::ReadChangelogFile()’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:259:46,
    inlined from ‘ChangelogWindow::SetPersonality(int)’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:97:39,
    inlined from ‘WindowChangelogOpen(int) [clone .part.0]’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:317:34:
/usr/include/c++/12/streambuf:490:30: error: potential null pointer dereference [-Werror=null-dereference]
  490 |       gptr()  const { return _M_in_cur;  }
      |                              ^~~~~~~~~
In member function ‘std::basic_streambuf<char, std::char_traits<char> >::egptr() const’,
    inlined from ‘std::basic_streambuf<char, std::char_traits<char> >::sbumpc()’ at /usr/include/c++/12/streambuf:324:49,
    inlined from ‘std::istreambuf_iterator<char, std::char_traits<char> >::operator++()’ at /usr/include/c++/12/bits/streambuf_iterator.h:172:17,
    inlined from ‘std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct<std::istreambuf_iterator<char, std::char_traits<char> > >(std::istreambuf_iterator<char, std::char_traits<char> >, std::istreambuf_iterator<char, std::char_traits<char> >, std::input_iterator_tag)void’ at /usr/include/c++/12/bits/basic_string.tcc:205:6,
    inlined from ‘std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string<std::istreambuf_iterator<char, std::char_traits<char> >, void>(std::istreambuf_iterator<char, std::char_traits<char> >, std::istreambuf_iterator<char, std::char_traits<char> >, std::allocator<char> const&)’ at /usr/include/c++/12/bits/basic_string.h:756:16,
    inlined from ‘ChangelogWindow::GetChangelogText[abi:cxx11]()’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:74:98,
    inlined from ‘ChangelogWindow::ReadChangelogFile()’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:259:46,
    inlined from ‘ChangelogWindow::SetPersonality(int)’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:97:39,
    inlined from ‘WindowChangelogOpen(int) [clone .part.0]’ at /tmp/OpenRCT2/src/openrct2-ui/windows/Changelog.cpp:317:34:
/usr/include/c++/12/streambuf:493:30: error: potential null pointer dereference [-Werror=null-dereference]
  493 |       egptr() const { return _M_in_end; }
      |                              ^~~~~~~~~
```

It seems like `-Wno-error=null-dereference` is needed but `-Wno-error=restrict` can be dropped.  I just force-pushed a revised commit dropping it.